sdk/appconfiguration/azure-data-appconfiguration/src/test/java/com/azure/data/appconfiguration/ConfigurationClientTest.java

```diff
@@ -70,13 +71,15 @@ private ConfigurationClient getConfigurationClient(HttpClient httpClient,
         return clientSetup(credentials -> {
             ConfigurationClientBuilder builder = new ConfigurationClientBuilder()
                 .connectionString(connectionString)
-                .httpClient(buildSyncAssertingClient(httpClient == null ? interceptorManager.getPlaybackClient() : httpClient))
+                .httpClient(buildSyncAssertingClient(interceptorManager.isPlaybackMode() ? interceptorManager.getPlaybackClient() : httpClient))
```

Comment:
Call `interceptorManager.setHttpClient` with the passed in HTTP client since you're paramaterizing things?

Reply:
I believe `interceptorManager.setHttpClient` is a private method. 
And `interceptorManager.getPlaybackClient()` is using the parameterized client underneath.

Reply:
Ah, yeah. My mistake. You can back that part of the change out, sorry. (I think we still need something like this to enable true multi-client testing, but this isn't the right API shape for it anyway. It should just be you pass an `HttpClient` into `getPlaybackClient`, I think.)